### PR TITLE
LPD-17297 Allow workflow admins to use Role-type reassignments through the UI timers configuration

### PR DESCRIPTION
**Pull request by pedro-oliveira446**
[LPD-17297](https://liferay.atlassian.net/browse/LPD-17297)

### CODE REVIEW

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Review by cgmonte — COMMENTED** [review]
Hey, thanks for the great work, as always, @pedro-oliveira446! I made some small change requests and a general comment.

As we talked in person, there is also bug preventing users from deleting the RoleType reassignment subsections. After we click new section and add a new RoleType reassignment, no trash can icon is shown. Please, check the comments for a possible solution.

File: modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/RoleType.js [hunk 1]
```diff
@@ -30,14 +30,20 @@ const RoleType = (props) => {
...
+	const {autoCreate, roleKey, roleName, roleType} = props.restProps;
+
 	return (
 		<SidebarPanel panelTitle={Liferay.Language.get('selected-role')}>
 			<BaseRoleType
+				autoCreate={autoCreate}
 				buttonName={Liferay.Language.get('new-section')}
 				inputLabel={Liferay.Language.get('role-type')}
 				networkStatus={networkStatus}
 				{...props}
```
[the remark this answers is not in the document]
Comment (cgmonte) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/RoleType.js:69]: It turns out the order is relevant to fix the bug we talked about. Please, check if the following change shows the trash can icon and makes it functional:

```
		<SidebarPanel panelTitle={Liferay.Language.get('selected-role')}>
			<BaseRoleType
				{...props}
				accountRoles={accountRoles}
				autoCreate={autoCreate}
				buttonName={Liferay.Language.get('new-section')}
				identifier={props.subSectionIdentifier}
				inputLabel={Liferay.Language.get('role-type')}
				networkStatus={networkStatus}
				resource={resource}
				roleKey={roleKey}
				roleName={roleName}
				roleType={roleType}
				sectionsLength={props.subSectionsLength}
			/>
		</SidebarPanel>
```

[the remark this answers is not in the document]
Comment (cgmonte) [reply modules/dxp/apps/portal-workflow/portal-workflow-kaleo-designer-web/src/main/resources/META-INF/resources/designer/js/definition-builder/diagram-builder/components/sidebar/sections/timers/select-reassignment/RoleType.js:69]: One (hopefully) last addition to my comment:

It is better to do a destructuring assignment of the component's properties, like this:

```
const RoleType = ({subSectionIdentifier, subSectionsLength, ...otherProps}) => {
...
}
```

This is more in line with the patterns we see in the Portal.

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Review by cgmonte — APPROVED** [review]
LGTM!

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:workflow

**Comment by pedro-oliveira446** [comment]
ci:workflow-metrics-test

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
ci:test:sf

**Comment by pedro-oliveira446** [comment]
ci:test:relevant

**Comment by pedro-oliveira446** [comment]
Resent : https://github.com/liferay-workflow/liferay-portal/pull/1102

**Pull request closed without merging**